src/Chunk.ts

```diff
@@ -673,160 +688,244 @@ export default class Chunk {
-					(code: string) => {
-						let map: SourceMap;
+			const namespace = module.namespace();
+			if (namespace.needsNamespaceBlock || !source.isEmpty()) {
+				magicString.addSource(source);
+				this.usedModules.push(module);
+
+				if (namespace.needsNamespaceBlock) {
+					const rendered = namespace.renderBlock(renderOptions);
+					if (namespace.renderFirst()) hoistedSource += '\n' + rendered;
+					else magicString.addSource(new MagicString(rendered));
+				}
+			}
+		}
 
-						if (options.sourcemap) {
-							timeStart('sourcemap', 3);
+		// resolve external module paths
+		for (let external of this.dependencies) {
+			if (!(external instanceof ExternalModule)) continue;
+			external.setRenderPath(options, this.entryModule && this.entryModule.id);
+		}
 
-							let file = options.file ? options.sourcemapFile || options.file : this.id;
-							if (file) file = resolve(typeof process !== 'undefined' ? process.cwd() : '', file);
+		if (hoistedSource) magicString.prepend(hoistedSource + '\n\n');
 
-							if (
-								this.graph.hasLoaders ||
-								this.graph.plugins.find(plugin =>
-									Boolean(plugin.transform || plugin.transformBundle)
-								)
-							) {
-								let decodedMap = magicString.generateDecodedMap({});
-								map = collapseSourcemaps(this, file, decodedMap, usedModules, bundleSourcemapChain);
-							} else {
-								map = magicString.generateMap({ file, includeContent: true });
-							}
+		this.renderedSource = magicString.trim();
 
-							map.sources = map.sources.map(normalize);
+		if (
+			this.getExportNames().length === 0 &&
+			this.getImportIds().length === 0 &&
+			this.renderedSource.isEmpty()
+		) {
+			this.graph.warn({
+				code: 'EMPTY_BUNDLE',
+				message: 'Generated an empty bundle'
+			});
+		}
 
-							timeEnd('sourcemap', 3);
-						}
+		this.hasDynamicImport = !!renderOptions.importMechanism;
+
+		this.renderedDeclarations = {
+			dependencies: this.getChunkDependencyDeclarations(options),
+			exports: this.getChunkExportDeclarations()
+		};
+	}
+
+	generateNamePreserveModules(preserveModulesRelativeDir: string) {
+		return (this.id = normalize(relative(preserveModulesRelativeDir, this.entryModule.id)));
+	}
 
-						if (code[code.length - 1] !== '\n') code += '\n';
-						return { code, map };
+	generateName(pattern: string, addons: Addons, existingNames?: { [name: string]: boolean }) {
+		// replace any chunk replacements
+		let outName = pattern.replace(/\[(hash|alias)\]/g, type => {
+			switch (type) {
+				case '[hash]':
+					return this.computeFullHash(addons);
+				case '[alias]':
+					if (!this.isEntryModuleFacade) {
+						return 'chunk';
+					} else if (this.entryModule.alias) {
+						return this.entryModule.alias;
+					} else {
+						let alias = basename(this.entryModule.id);
+						if (alias.endsWith('.js')) alias = alias.substr(0, alias.length - 3);
+						else if (alias.endsWith('.mjs')) alias = alias.substr(0, alias.length - 4);
+						return alias;
```

Comment:
Some developers like to use dots in paths to indicate hierarchy - `page.component.thing`. Assuming extensions for `.` will affect these kinds of scenarios. Also the `alias` shouldn't be considered itself a file path.

In terms of validating the `alias` further it could be worth thinking about ensuring it doesn't support `..`, `.` or `\` characters etc. with some validation.